internal/guest/storage/scsi/scsi.go

```diff
@@ -282,8 +293,38 @@ func ControllerLunToName(ctx context.Context, controller, lun uint8) (_ string,
 	if len(deviceNames) > 1 {
 		return "", errors.Errorf("more than one block device could match SCSI ID \"%s\"", scsiID)
 	}
+	deviceName := deviceNames[0].Name()
+
+	// devices that have partitions have a subdirectory under
+	// /sys/bus/scsi/devices/<scsiID>/block/<deviceName> for each partition.
+	// Partitions use 1-based indexing, so if `partition` is 0, then we should
+	// return the device name without a partition index.
+	if partition != 0 {
+		partitionName := fmt.Sprintf("%s%d", deviceName, partition)
+		partitionPath := filepath.Join(blockPath, deviceName, partitionName)
+
+		// Wait for the device partition to show up
+		for {
+			fi, err := osStat(partitionPath)
+			if err != nil && !errors.Is(err, fs.ErrNotExist) {
+				return "", err
+			} else if fi == nil {
+				// if the fileinfo is nil that means we didn't find the device, keep
+				// trying until the context is done or the device path shows up
+				select {
+				case <-ctx.Done():
+					return "", ctx.Err()
+				default:
+					time.Sleep(time.Millisecond * 10)
+					continue
+				}
+			}
+			break
+		}
+		deviceName = partitionName
+	}
 
-	devicePath := filepath.Join("/dev", deviceNames[0].Name())
+	devicePath := filepath.Join("/dev", deviceName)
```

Comment:
so it's `/dev/sda1`, `/dev/sda2` etc?

Reply:
Yep! Or if the device is not partitioned, then it would be the normal `/dev/sda`.